plugins/modules/vmware_vmkernel_ip_config.py

```diff
@@ -89,7 +105,8 @@ def main():
     argument_spec = vmware_argument_spec()
     argument_spec.update(dict(vmk_name=dict(required=True, type='str'),
                               ip_address=dict(required=True, type='str'),
-                              subnet_mask=dict(required=True, type='str')))
+                              subnet_mask=dict(required=True, type='str'),
+                              gateway=dict(required=True, type='str')))
```

Comment:
```suggestion
                              gateway=dict(type='str')))
```
Please don't make this a required parameter. We've been running dedicated vMotion vmk's without a gateway (the ESXi hosts are in the same subnet and can talk to each other directly) for years without any problems, so obviously there are use cases where you _don't_ need one.